[Woo POS] Fix status bar height handling in Orders Loading and Order Details screen

samiuelson opened the pull request:
<!-- Remember about a good descriptive title. -->

<!-- Id number of the Linear issue this PR addresses, e.g., WOOMOB-373. -->
WOOMOB-1809

### Description
<!-- Take the time to write a good summary. Why is it needed? What does it do? When fixing bugs try to avoid just writing “See original issue” – clarify what the problem was and how you’ve fixed it. -->

This pull request fixes status bar height handling in the WooCommerce POS Orders Loading and Order Details screens by applying proper status bar padding to ensure UI elements are not obscured by the system status bar.

- Renamed WooPosOrdersLoadingState to WooPosOrdersLoadingScreen for consistency
- Applied statusBarsPadding() modifier to both loading and detail screens to account for status bar height
- Adjusted padding and alignment in order detail views to maintain proper spacing with the new status bar handling

### Test Steps
<!-- Describe how to test your changes. Include only what’s needed for the reviewer to validate the behavior.
For new features, outline the main user flow or goal rather than every tap or click — the reviewer should be able to complete the flow naturally.
If applicable, mention key devices, scenarios, or edge cases to verify. -->

Verify POS Orders Loading and Content state have correct UI.

### Images/gif
<!-- Include before and after images or gifs when appropriate. -->
#### Loading
| before | after |
|---|---|
|<img width="500" alt="loading-before" src="https://github.com/user-attachments/assets/14bc6c52-3e4b-4615-91d5-6372d3637374" />|<img width="500" alt="loading-after" src="https://github.com/user-attachments/assets/bb2bc50b-02a9-451e-b746-e7d889821110" />|

#### Content
| before | after |
|---|---|
|<img width="500"  alt="loaded-before" src="https://github.com/user-attachments/assets/5eba2dc2-5354-458b-9d00-2ce0a4b7ba93" />|<img width="500" alt="loaded-after" src="https://github.com/user-attachments/assets/c1ee5e6f-a480-4e22-bb06-a5d8712ad4a3" />|

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

<!-- Pull request guidelines: https://github.com/woocommerce/woocommerce-android/blob/develop/docs/pull-request-guidelines.md -->

samiuelson committed:
Update layout/spacing in WooPosOrdersLoadingState

- Add statusBarsPadding to the root Row to respect system windows.
- Remove top padding from WooPosOrderDetailsLoadingPane modifier.
- Update padding in WooPosOrderDetailsLoadingPane to be horizontal-only.
- Add a Spacer and vertical padding for better visual alignment in the details pane.

samiuelson committed:
Update WooPosOrdersDetails layout with status bar padding and toolbar height

- Add `statusBarsPadding()` to the main column modifier to prevent overlap with system bars.
- Set a minimum height for the toolbar row using `WOO_POS_ORDERS_TOOLBAR_HEIGHT`.
- Align toolbar content vertically to the center.

samiuelson committed:
Rename WooPosOrdersLoadingState to WooPosOrdersLoadingScreen

samiuelson commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/orders/WooPosOrdersLoadingScreen.kt:
```diff
@@ -40,8 +41,11 @@ import com.woocommerce.android.ui.woopos.common.composeui.designsystem.WooPosThe
 import com.woocommerce.android.ui.woopos.common.composeui.designsystem.WooPosTypography
 
 @Composable
-fun WooPosOrdersLoadingState() {
```
`WooPosOrdersLoadingState` was a bit misleading, because it fits more of a name for a UI state model.

kidinov replied:
@samueljseay 

Screen is also misleading, as this is not a screen but one of the "states" (loading) of the OrdersScreen

samiuelson committed:
Satisfy detekt's complaints

kidinov commented on WooCommerce/src/main/kotlin/com/woocommerce/android/ui/woopos/orders/WooPosOrdersLoadingScreen.kt:
```diff
@@ -40,8 +41,12 @@ import com.woocommerce.android.ui.woopos.common.composeui.designsystem.WooPosThe
+    Row(
+        modifier = Modifier
+            .fillMaxSize()
+            .statusBarsPadding()
```
Shell we have it on the top level `WooPosOrdersScreen`? Why only 2 states not the rest then?

kidinov reviewed (commented):
@samiuelson I think it's not the best handling of this.

* Empty and error states not handled
* Loading state has the layout broken - notice the "list" part has padding at the top when loading.

<img width="1549" height="973" alt="Image" src="https://github.com/user-attachments/assets/a328cb42-b23d-4152-9621-2aa23c8ebb17" />

samiuelson committed:
Apply statusBarsPadding to the main Box in WooPosOrdersScreen and remove redundant paddings

Refactor the layout structure in WooPosOrdersScreen to apply `statusBarsPadding()` at the root Box level instead of on individual child components like the search bar. Remove redundant `statusBarsPadding` and adjust padding values in `WooPosOrdersDetails` and `WooPosOrdersLoadingScreen` to align with this change.

kidinov reviewed (changes requested):
@samiuelson 

Now the issue I mentioned in the previous review both on the content and loading states

<img width="558" height="353" alt="Image" src="https://github.com/user-attachments/assets/97b16e78-4bbf-40cb-a8c0-f5ed90a4dae1" />

samiuelson committed:
Fix status bar insets in WooPos Orders screens

- Remove global `statusBarsPadding` from the root `WooPosOrdersScreen` container.
- Apply `statusBarsPadding` individually to `OrdersContent`, `OrdersEmpty`, `OrdersError`, and `WooPosToolbar`.
- Add `statusBarsPadding` to `WooPosOrdersDetails`.
- Refactor `WooPosOrdersLoadingScreen` layout to correctly handle status bar insets and toolbar spacing.

kidinov reviewed (changes requested):
@samiuelson Looks close, but there is still an issue - the loading is not aligned with content and "jumping"

https://github.com/user-attachments/assets/ca7ad5d2-96b0-4720-a718-ef6839b718cf

<img width="1088" height="108" alt="Image" src="https://github.com/user-attachments/assets/91885c7f-c5cd-4496-b532-6020e8fb4d87" />

<img width="1089" height="101" alt="Image" src="https://github.com/user-attachments/assets/3ad4f68f-0b84-4875-b0d1-30ee032c814d" />

Its not the case on the trunk

samiuelson committed:
Merge branch 'trunk' into pos-orders-fix-status-bar-handling

samiuelson committed:
Stop jumping

kidinov reviewed (changes requested):
@samiuelson 👋

Now the loading list has way to big margin at the top


https://github.com/user-attachments/assets/3fd75b9e-ce4c-4d32-8162-d58dc5941b29

<img width="244" height="196" alt="image" src="https://github.com/user-attachments/assets/286eac1c-c486-4d2c-a968-e71615842260" />

samiuelson committed:
Fix space size when loading orders search results

samiuelson commented:
> Now the loading list has way to big margin at the top

Good catch, @kidinov! Fixed: fea4682

<img width="800" alt="Screenshot_20251203_114554" src="https://github.com/user-attachments/assets/b6020bba-97a5-461a-9807-b1464ff3b3db" />

samiuelson committed:
Merge branch 'trunk' into pos-orders-fix-status-bar-handling

kidinov reviewed (changes requested):
@samiuelson 👋

Now we are back to padding at the top of the orders section on the loading state

https://github.com/user-attachments/assets/92500903-d9df-4699-9e6e-06df414198da

<img width="535" height="297" alt="Image" src="https://github.com/user-attachments/assets/cc987f6d-cbbb-4430-abc5-bd3befbaf5ad" />

samiuelson committed:
Draw background behind status bar when loading

kidinov reviewed (approved):
LGTM!
